powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/service/v3/ActivationService.java

```diff
@@ -265,22 +265,22 @@ public ActivationLayer1Response createActivation(ActivationLayer1Request request
                 }
 
                 default:
-                    throw new PowerAuthAuthenticationException("Unsupported activation type: " + request.getType());
+                    throw new PowerAuthAuthenticationException("POWER_AUTH_REQUEST_INVALID");
             }
         } catch (PowerAuthClientException ex) {
             if (ex.getPowerAuthError() instanceof PowerAuthErrorRecovery) {
                 PowerAuthErrorRecovery errorRecovery = (PowerAuthErrorRecovery) ex.getPowerAuthError();
                 throw new PowerAuthRecoveryException(ex.getMessage(), "INVALID_RECOVERY_CODE", errorRecovery.getCurrentRecoveryPukIndex());
             }
```

Comment:
Let's pass `ex` as a parameter to the new exception to keep the cause somewhere.